diff --git a/node_modules/torrent-search-api/lib/providers/1337x.js b/node_modules/torrent-search-api/lib/providers/1337x.js
index 9d50b64..b4860ae 100755
--- a/node_modules/torrent-search-api/lib/providers/1337x.js
+++ b/node_modules/torrent-search-api/lib/providers/1337x.js
@@ -6,15 +6,15 @@ class _1337x extends TorrentProvider {
     super({
       name: '1337x',
       baseUrl: 'https://www.1337x.to',
-      searchUrl: '/category-search/{query}/{cat}/1/',
+      searchUrl: '/sort-category-search/{query}/{cat}/{sort}/desc/1/', // https://1337x.to/sort-category-search/discography/Music/time/desc/1/
       categories: {
-        All: 'url:/search/{query}/1/',
+        All: 'url:/sort-search/{query}/{sort}/desc/1/', // https://www.1337x.to/sort-search/discography/time/desc/1/
         Movies: 'Movies',
         TV: 'TV',
         Games: 'Games',
         Music: 'Music',
         Anime: 'Anime',
-        Applications: 'Apps',
+        Apps: 'Apps',
         Documentaries: 'Documentaries',
         Xxx: 'XXX',
         Other: 'Other',
@@ -25,6 +25,7 @@ class _1337x extends TorrentProvider {
       itemsSelector: 'tbody > tr',
       itemSelectors: {
         title: 'a:nth-child(2)',
+        category: 'a:nth-child(1) > i@class',
         time: '.coll-date',
         seeds: '.seeds | int',
         peers: '.leeches | int',
@@ -37,11 +38,11 @@ class _1337x extends TorrentProvider {
     });
   }
 
-  downloadTorrent(torrent,path) {
+  downloadTorrent(torrent, path) {
     return this.xray(torrent.desc, '.infohash-box span').then(hash =>
       super.downloadTorrent({
         link: format('http://itorrents.org/torrent/{0}.torrent', hash)
-      },path)
+      }, path)
     );
   }
 }
diff --git a/node_modules/torrent-search-api/lib/providers/eztv.js b/node_modules/torrent-search-api/lib/providers/eztv.js
new file mode 100644
index 0000000..ea4a442
--- /dev/null
+++ b/node_modules/torrent-search-api/lib/providers/eztv.js
@@ -0,0 +1,40 @@
+const TorrentProvider = require('../TorrentProvider');
+
+class Eztv extends TorrentProvider {
+  constructor() {
+    super({
+      name: 'Eztv',
+      baseUrl: 'https://eztv.re',
+      searchUrl: '/search/{query}',
+      categories: {
+        All: ''
+      },
+      defaultCategory: 'All',
+      resultsPerPageCount: 50,
+      itemsSelector: 'tr.forum_header_border',
+      itemSelectors: {
+        title: 'a.epinfo@text',
+        link: 'a.download_1@href',
+        magnet: 'a.magnet@href',
+        time: 'td:nth-last-child(2)@text',
+        seeds: 'td:nth-last-child(1) | int',
+        size: 'td:nth-last-child(3)@text',
+        desc: 'a.epinfo@href'
+      },
+      paginateSelector: 'a:contains( next page)@href',
+      torrentDetailsSelector: '.section_thread_post@html'
+    });
+  }
+
+  postProcess(results) {
+    return results.map(r => {
+      // EZTV doesn't provide peer counts, explicitly set to null
+      r.peers = null;
+      r.provider = this.name;
+      return r;
+    });
+  }
+}
+
+module.exports = Eztv;
+
diff --git a/node_modules/torrent-search-api/lib/providers/eztv.json b/node_modules/torrent-search-api/lib/providers/eztv.json
deleted file mode 100644
index f47fd2d..0000000
--- a/node_modules/torrent-search-api/lib/providers/eztv.json
+++ /dev/null
@@ -1,22 +0,0 @@
-{
-    "name": "Eztv",
-    "baseUrl": "https://eztv.re",
-    "searchUrl": "/search/{query}",
-    "categories": {
-       "All": ""
-    },
-    "defaultCategory": "All",
-    "resultsPerPageCount": 50,
-    "itemsSelector": "tr.forum_header_border",
-    "itemSelectors": {
-       "title": "a.epinfo@text",
-       "link": "a.download_1@href",
-       "magnet": "a.magnet@href",
-       "time": "td.forum_thread_post:nth-child(5)@text",
-       "seeds": "td.forum_thread_post_end | int",
-       "size": "td.forum_thread_post:nth-child(4)@text",
-       "desc": "a.epinfo@href"
-    },
-    "paginateSelector": "a:contains( next page)@href",
-    "torrentDetailsSelector": ".section_thread_post@html"
- }
\ No newline at end of file
diff --git a/node_modules/torrent-search-api/lib/providers/kickasstorrents.json b/node_modules/torrent-search-api/lib/providers/kickasstorrents.json
index aa954ff..ee48ecc 100755
--- a/node_modules/torrent-search-api/lib/providers/kickasstorrents.json
+++ b/node_modules/torrent-search-api/lib/providers/kickasstorrents.json
@@ -1,29 +1,29 @@
 {
-    "name": "KickassTorrents",
-    "baseUrl": "https://katcr.co",
-    "searchUrl": "/katsearch/page/1/{query}",
-    "categories": {
-       "All": "",
-       "Movies": "url:/category/movies/page/1",
-       "TV": "url:/category/tv/page/1",
-       "Music": "url:/category/music/page/1",
-       "Games": "url:/category/games/page/1",
-       "Books": "url:/category/books/page/1",
-       "Applications": "url:/category/applications/page/1",
-       "Anime": "url:/category/anime/page/1"
-    },
-    "defaultCategory": "All",
-    "resultsPerPageCount": 25,
-    "itemsSelector": "tbody tr",
-    "itemSelectors": {
-      "title": ".torrents_table__torrent_title@text | trim",
+   "name": "KickassTorrents",
+   "baseUrl": "https://katcr.to",
+   "searchUrl": "/usearch/{query}/",
+   "categories": {
+      "All": "",
+      "Movies": "url:{query}/category/movies/1",
+      "TV": "url:/category/tv/",
+      "Music": "url:/category/music/1",
+      "Games": "url:/category/games/1",
+      "Books": "url:/category/books/1",
+      "Applications": "url:/category/applications/1",
+      "Anime": "url:/category/anime/1"
+   },
+   "defaultCategory": "All",
+   "resultsPerPageCount": 25,
+   "itemsSelector": ".data tbody tr.odd",
+   "itemSelectors": {
+      "title": ".torrentname@text | trim",
       "time": "td[data-title=\"Age\"]",
       "seeds": "td[data-title=\"Seed\"] | int",
       "peers": "td[data-title=\"Leech\"] | int",
       "size": "td[data-title=\"Size\"]",
       "link": "a[title=\"Download torrent file\"]@href",
       "desc": ".torrents_table__torrent_title@href"
-    },
-    "paginateSelector": "a:contains(Next)@href",
-    "torrentDetailsSelector": ".col@html"
- }
\ No newline at end of file
+   },
+   "paginateSelector": "a:contains(Next)@href",
+   "torrentDetailsSelector": ".col@html"
+}
\ No newline at end of file
diff --git a/node_modules/torrent-search-api/lib/providers/limetorrents.json b/node_modules/torrent-search-api/lib/providers/limetorrents.json
index f35ce62..d310621 100644
--- a/node_modules/torrent-search-api/lib/providers/limetorrents.json
+++ b/node_modules/torrent-search-api/lib/providers/limetorrents.json
@@ -1,6 +1,6 @@
 {
     "name": "Limetorrents",
-    "baseUrl": "https://www.limetorrents.info",
+    "baseUrl": "https://www.limetorrents.lol",
     "enableCloudFareBypass": true,
     "searchUrl": "/search/{cat}/{query}/seeds/1/",
     "categories": {
@@ -16,14 +16,14 @@
     "resultsPerPageCount": 40,
     "itemsSelector": ".table2 tr",
     "itemSelectors": {
-       "title": ".tt-name > a:nth-child(2)@text",
-       "link": "a.csprite_dl14@href",
-       "seeds": ".tdseed | int",
-       "peers": ".tdleech | int",
-       "time": "td.tdnormal:nth-child(2)@text  | match:\"(.+?) -\"",
-       "size": "td.tdnormal:nth-child(3)@text",
-       "desc": ".tt-name > a:nth-child(2)@href"
+        "title": ".tt-name > a:nth-child(2)@text",
+        "link": "a.csprite_dl14@href",
+        "seeds": ".tdseed | int",
+        "peers": ".tdleech | int",
+        "time": "td.tdnormal:nth-child(2)@text  | match:\"(.+?) -\"",
+        "size": "td.tdnormal:nth-child(3)@text",
+        "desc": ".tt-name > a:nth-child(2)@href"
     },
     "paginateSelector": "a:contains(Next page)@href",
     "torrentDetailsSelector": "#content@html"
- }
\ No newline at end of file
+}
\ No newline at end of file
diff --git a/node_modules/torrent-search-api/lib/providers/rarbg.js b/node_modules/torrent-search-api/lib/providers/rarbg.js
index df88f4a..2a1cfa7 100755
--- a/node_modules/torrent-search-api/lib/providers/rarbg.js
+++ b/node_modules/torrent-search-api/lib/providers/rarbg.js
@@ -1,86 +1,36 @@
-const util = require('util');
-const request = util.promisify(require('request'));
 const TorrentProvider = require('../TorrentProvider');
-const { humanizeSize } = require('../utils/helpers');
 
 class Rarbg extends TorrentProvider {
   constructor() {
     super({
       name: 'Rarbg',
-      baseUrl: 'https://torrentapi.org',
-      getTokenUrl:
-        '/pubapi_v2.php?get_token=get_token&app_id=NodeTorrentSearchApi',
-      searchUrl:
-        '/pubapi_v2.php?app_id=NodeTorrentSearchApi&search_string={query}&category={cat}&mode=search&format=json_extended&sort=seeders&limit=100&token=',
+      baseUrl: 'https://rargb.to',
+      searchUrl: '/search/?search={query}',
       categories: {
-        All:
-          '1;4;14;15;16;17;21;22;42;18;19;41;27;28;29;30;31;32;40;23;24;25;26;33;34;43;44;45;46;47;48;49;50;51;52',
-        Movies: '14;17;42;44;45;46;47;48;50;51;52',
-        XXX: '1;4',
-        Games: '1;27;28;29;30;31;32;40',
-        TV: '1;18;41;49',
-        Music: '1;23;24;25;26',
-        Apps: '1;33;34;43',
-        Books: '35'
+        All: '',
+        Movies: '&category%5B%5D=14&category%5B%5D=48&category%5B%5D=17&category%5B%5D=44&category%5B%5D=45&category%5B%5D=47&category%5B%5D=50&category%5B%5D=51&category%5B%5D=42&category%5B%5D=46&category%5B%5D=52&category%5B%5D=54',
+        TV: '&category%5B%5D=18&category%5B%5D=41&category%5B%5D=49',
+        Music: '&category%5B%5D=23&category%5B%5D=24&category%5B%5D=25&category%5B%5D=26',
+        Games: '&category%5B%5D=27&category%5B%5D=28&category%5B%5D=29&category%5B%5D=30&category%5B%5D=31&category%5B%5D=32&category%5B%5D=40&category%5B%5D=53',
+        Apps: '&category%5B%5D=33&category%5B%5D=34&category%5B%5D=43',
+        XXX: '&category%5B%5D=4'
       },
       defaultCategory: 'All',
-      resultsPerPageCount: 100
+      resultsPerPageCount: 25,
+      itemsSelector: 'table.lista2t tr.lista2',
+      itemSelectors: {
+        title: 'td:nth-child(2) a[href*="/torrent/"]',
+        time: 'td:nth-child(4)',
+        size: 'td:nth-child(5)',
+        seeds: 'td:nth-child(6) | int',
+        peers: 'td:nth-child(7) | int',
+        desc: 'td:nth-child(2) a[href*="/torrent/"]@href'
+      },
+      paginateSelector: 'a[href*="/search/"][href*="page="]@href | last',
+      magnetSelector: 'a[href^="magnet:"]@href',
+      enableCloudFareBypass: false
     });
   }
-
-  search(query, category) {
-    return this.ensureLogin()
-      .then(() => {
-        const url = this.getUrl(category, query);
-        if (url === null) {
-          return Promise.resolve([]);
-        }
-        return request({ url, headers: { 'User-Agent': 'curl/7.37.0' } });
-      })
-      .then(response => {
-        const results = JSON.parse(response.body).torrent_results;
-        return (
-          results &&
-          results.map(r => ({
-            provider: this.name,
-            title: r.title,
-            time: r.pubdate,
-            seeds: r.seeders,
-            peers: r.leechers,
-            size: humanizeSize(r.size, r.name),
-            magnet: r.download,
-            desc: r.info_page
-          }))
-        );
-      });
-  }
-
-  getTorrentDetails(torrent) {
-    throw new Error('Not implemented');
-  }
-
-  ensureLogin() {
-    if (!this.lastLoginTime || Date.now() - this.lastLoginTime > 840000) {
-      return request({
-        url: this.baseUrl + this.getTokenUrl,
-        headers: { 'User-Agent': 'curl/7.37.0' }
-      }).then(r => {
-        //wait 2 seconds to avoid doing more than 1 req per 2 secs
-        const waitTill = new Date(new Date().getTime() + 2 * 1000);
-        while (waitTill > new Date()) {}
-
-        this.lastLoginTime = Date.now();
-        this.searchUrl =
-          this.searchUrl.substr(0, this.searchUrl.lastIndexOf('=') + 1) +
-          JSON.parse(r.body).token;
-      });
-    }
-    return Promise.resolve();
-  }
-
-  downloadTorrent() {
-    throw new Error('Not implemented');
-  }
 }
 
 module.exports = Rarbg;
diff --git a/node_modules/torrent-search-api/lib/providers/torrent9.js b/node_modules/torrent-search-api/lib/providers/torrent9.js
index 0aa06c9..e20f7dd 100644
--- a/node_modules/torrent-search-api/lib/providers/torrent9.js
+++ b/node_modules/torrent-search-api/lib/providers/torrent9.js
@@ -4,16 +4,17 @@ class Torrent9 extends TorrentProvider {
   constructor() {
     super({
       name: 'Torrent9',
-      baseUrl: 'https://www.torrent9.ninja',
+      baseUrl: 'https://torrent9.to/',
       enableCloudFareBypass: true,
-      searchUrl: '/recherche/{query}',
+      searchUrl: '/search_torrent/{query}',
       categories: {
         All: '',
-        Movies: 'url:/torrents/films',
-        TV: 'url:/torrents/series',
-        Music: 'url:/torrents/musique',
-        Apps: 'url:/torrents/logiciels',
-        Books: 'url:/torrents/ebook',
+        Movies: 'url:/search_torrent/films/{query}.html',
+        TV: 'url:/search_torrent/series/{query}.html',
+        Music: 'url:/search_torrent/musique/{query}.html',
+        Apps: 'url:/search_torrent/logiciels/{query}.html',
+        Books: 'url:/search_torrent/ebook/{query}.html',
+        Games: 'url:/search_torrent/jeux-pc/{query}.html',
         Top100: 'url:/top'
       },
       defaultCategory: 'All',
@@ -21,10 +22,12 @@ class Torrent9 extends TorrentProvider {
       itemsSelector: 'table tr',
       itemSelectors: {
         title: 'a',
+        time: 'td:nth-child(2)',
         seeds: '.seed_ok | int',
         peers: 'td:nth-child(4) | int',
-        size: 'td:nth-child(2)',
-        desc: 'a@href | replace:t//,t/'
+        size: 'td:nth-child(3)',
+        desc: 'a@href | replace:t//,t/',
+        category: 'i@class'
       },
       paginateSelector: 'a:contains(Suivant ►)@href',
       torrentDetailsSelector: '.movie-detail > .row:nth-child(1)@html'
diff --git a/node_modules/torrent-search-api/lib/providers/torrentz2.json b/node_modules/torrent-search-api/lib/providers/torrentz2.json
index d2a0656..3c88a42 100755
--- a/node_modules/torrent-search-api/lib/providers/torrentz2.json
+++ b/node_modules/torrent-search-api/lib/providers/torrentz2.json
@@ -1,24 +1,24 @@
 {
-  "name": "Torrentz2",
-  "baseUrl": "https://torrentz2.eu",
-  "searchUrl": "/?f={query}",
-  "categories": {
-     "All": ""
-  },
-  "defaultCategory": "All",
-  "resultsPerPageCount": 50,
-  "itemsSelector": "div.results dl",
-  "itemSelectors": {
-     "title": "a",
-     "hash": "a@href | substr:21,",
-     "time": "span:nth-child(2)",
-     "size": "span:nth-child(3)",
-     "seeds": "span:nth-child(5) | int",
-     "peers": "span:nth-child(4) | int",
-     "magnet": "a@href | substr:21, | format:magnet:?xt=urn:btih:{0}",
-     "link": "a@href | substr:21, | format:http://itorrents.org/torrent/{0}.torrent"
-  },
-  "paginateSelector": "a:contains(»)@href",
-  "torrentDetailsSelector": "td[class*=\"frame\"]:nth-child(1)@html",
-  "enableCloudFareBypass": true
+   "name": "Torrentz2",
+   "baseUrl": "https://torrentz2.nz",
+   "searchUrl": "/search?q={query}",
+   "categories": {
+      "All": ""
+   },
+   "defaultCategory": "All",
+   "resultsPerPageCount": 50,
+   "itemsSelector": "div.results dl",
+   "itemSelectors": {
+      "title": "a",
+      "hash": "a@href | substr:21,",
+      "time": "span:nth-child(2)",
+      "size": "span:nth-child(3)",
+      "seeds": "span:nth-child(5) | int",
+      "peers": "span:nth-child(4) | int",
+      "magnet": "a@href | substr:21, | format:magnet:?xt=urn:btih:{0}",
+      "link": "a@href | substr:21, | format:http://itorrents.org/torrent/{0}.torrent"
+   },
+   "paginateSelector": "a:contains(»)@href",
+   "torrentDetailsSelector": "td[class*=\"frame\"]:nth-child(1)@html",
+   "enableCloudFareBypass": true
 }
\ No newline at end of file
diff --git a/node_modules/torrent-search-api/lib/providers/yts.js b/node_modules/torrent-search-api/lib/providers/yts.js
index d649180..c562802 100644
--- a/node_modules/torrent-search-api/lib/providers/yts.js
+++ b/node_modules/torrent-search-api/lib/providers/yts.js
@@ -6,15 +6,15 @@ class Yts extends TorrentProvider {
   constructor() {
     super({
       name: 'Yts',
-      baseUrl: 'https://yts.mx',
+      baseUrl: 'https://yts.lt',
       searchUrl:
-        '/api/v2/list_movies.json?query_term={query}&sort=seeds&order=desc&set=1',
+        '/api/v2/list_movies.json?query_term={query}&sort=seeds&order=desc&limit=50',
       categories: {
         All: '',
         Movies: ''
       },
       defaultCategory: 'All',
-      resultsPerPageCount: 20
+      resultsPerPageCount: 50
     });
   }
 
