diff --git a/node_modules/torrent-search-api/lib/providers/yts.js b/node_modules/torrent-search-api/lib/providers/yts.js
index 4db6af1..c9f40aa 100644
--- a/node_modules/torrent-search-api/lib/providers/yts.js
+++ b/node_modules/torrent-search-api/lib/providers/yts.js
@@ -10,7 +10,8 @@ class Yts extends TorrentProvider {
       searchUrl:
         '/api/v2/list_movies.json?query_term={query}&sort=seeds&order=desc&set=1',
       categories: {
-        All: ''
+        All: '',
+        Movies: ''
       },
       defaultCategory: 'All',
       resultsPerPageCount: 20
